docs/book/src/clusterctl/developers.md

```diff
@@ -77,4 +77,59 @@ please note that each `provider_repo` should have its own `clusterctl-settings.j
     "nextVersion": "v0.5.0",
   }
 }
-```
\ No newline at end of file
+```
+
+## Additional steps in order to use the docker provider
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+The Docker provider is not designed for production use and is intended for development environments only.
+
+</aside>
+
+Before running the local-overrides hack:
+
+- Run `make -C test/infrastructure/docker docker-build REGISTRY=gcr.io/k8s-staging-capi-docker` to build the docker provider image
+  using a specific REGISTRY (you can choose your own).
+
+- Run `make -C test/infrastructure/docker generate-manifests REGISTRY=gcr.io/k8s-staging-capi-docker` to generate 
+  the docker provider manifest using the above registry/Image.
+
+Run the local-overrides hack.
+
+Edit the clusterctl config file located at `~/.cluster-api/clusterctl.yaml` and configure the docker provider
+by adding the following lines (replace $HOME with your home path):
+
+```bash
+providers:
+  - name: docker
+    url: $HOME/.cluster-api/overrides/infrastructure-docker/latest/infrastructure-components.yaml
```

Comment:
```
$ ./bin/clusterctl init --infrastructure=docker
Fetching providers
Error: failed to get provider components for the "docker" provider: failed to get repository client for the InfrastructureProvider with name docker: error creating the local filesystem repository client: invalid path: path "$HOME/.cluster-api/overrides/infrastructure-docker/latest/infrastructure-components.yaml" must be an absolute path
```

Reply:
@ncdc it is required to replace $HOME with your home path (it is described in the comment, might be it is not clear enough...)

Reply:
👍 I just copied the contents. That's on me.